tree/ntuple/v7/src/RNTupleSerialize.cxx

```diff
@@ -1635,3 +1759,40 @@ ROOT::Experimental::Internal::RNTupleSerializer::DeserializePageList(const void
 
    return RResult<void>::Success();
 }
+
+std::string ROOT::Experimental::Internal::RNTupleSerializer::SerializeStreamerInfos(const StreamerInfoMap_t &infos)
+{
+   TList streamerInfos;
+   for (auto si : infos) {
+      assert(si.first == si.second->GetNumber());
+      streamerInfos.Add(si.second);
+   }
+   TBufferFile buffer(TBuffer::kWrite);
+   buffer.WriteObject(&streamerInfos);
+   assert(buffer.Length() > 0);
+   return std::string{buffer.Buffer(), static_cast<UInt_t>(buffer.Length())};
+}
+
+ROOT::Experimental::RResult<ROOT::Experimental::Internal::RNTupleSerializer::StreamerInfoMap_t>
+ROOT::Experimental::Internal::RNTupleSerializer::DeserializeStreamerInfos(const std::string &extraTypeInfoContent)
+{
+   StreamerInfoMap_t infoMap;
+
+   TBufferFile buffer(TBuffer::kRead, extraTypeInfoContent.length(), const_cast<char *>(extraTypeInfoContent.data()),
+                      false /* adopt */);
+   auto infoList = reinterpret_cast<TList *>(buffer.ReadObject(TList::Class()));
+   infoList->SetOwner(); // delete the TStreamerInfo items of the list
+
+   TObjLink *lnk = infoList->FirstLink();
+   while (lnk) {
+      auto info = reinterpret_cast<TStreamerInfo *>(lnk->GetObject());
+      info->BuildCheck();
+      infoMap[info->GetNumber()] = info->GetClass()->GetStreamerInfo();
```

Comment:
Why is the assignment to the 'current' `StreamerInfo` rather that the one we are looking at?  Unless the input is guaranteed to be from this process, this seems wrong  ... and if it is guaranteed to be from this process, it seems wasteful to store and rebuild the whole `StreamerInfo` rather than just its name.  What am I missing?